common-util/List/IpfsHashGenerationModal/HashOfDataFile.jsx

```diff
@@ -36,52 +35,56 @@ const IpfsModal = ({
         },
         { noImage: true },
       );
-      if (callback) callback(hash);
-      onModalClose();
-
       return hash;
     } catch (error) {
       window.console.log(error);
     } finally {
-      setIsHashLoading(false); // off the loader and close the `Modal`
+      setIsHashLoading(false); // off the loader
     }
 
     return null;
   };
 
   const onFinish = async (values) => {
     const hash = await getNewHash(values);
-    if (callback) callback(hash);
+
+    handleSubmit({
+      ...values,
+      hash,
+    }, onModalClose);
   };
 
-  const handleOk = () => {
-    form.submit();
+  const onFinishFailed = (errorInfo) => {
+    console.log('Failed:', errorInfo); /* eslint-disable-line no-console */
   };
 
   return (
     <CustomModal
       open={visible}
-      title="Generate IPFS Hash of Metadata File"
+      title="New request"
       okText="Copy Hash & Close"
       cancelText="Cancel"
       destroyOnClose
       width={620}
       onCancel={handleCancel}
       footer={[
         <Fragment key="footer-1">
-          <Button type="default" onClick={onModalClose}>
-            Cancel
-          </Button>
-
-          <Button
-            form="myForm"
-            htmlType="submit"
-            type="primary"
-            loading={isHashLoading}
-            onClick={handleOk}
-          >
-            Save File & Generate Hash
-          </Button>
+          <div>
+            <Button type="default" onClick={onModalClose}>
+              Cancel
+            </Button>
+
+            <Button
+              form="ipfsModalForm"
+              htmlType="submit"
+              type="primary"
+              disabled={!account}
+              loading={isHashLoading || isLoading}
+            >
+              Request
+            </Button>
+          </div>
+          {!account && <div className="text-gray-500 mt-12">* Please connect your wallet</div>}
```

Comment:
```suggestion
          {!account && <div className="text-gray-500 mt-12">To make a request, connect your wallet</div>}
```

Reply:
Generally good to start the error with the name of the task the user is trying to achieve, followed by what they can do to fix.

Reply:
makes sense, fixed